src/TextInput/TextInput.tsx

```diff
@@ -144,7 +144,7 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
           onFocus={handleInputFocus}
           onBlur={handleInputBlur}
           type={type}
-          aria-required={required ? 'true' : 'false'}
+          aria-required={required}
```

Comment:
We still use `aria-required={required ? 'true' : 'false'}` on Radio and Checkbox. Should we update those as well? Or are we worried this could also break tests in dotcom?

Reply:
What are the broken tests?? 👀

Reply:
There are some [tests in dotcom that compare the dom](https://github.com/github/github/actions/runs/7176291085/job/19540971389), so any additional props even when false fail CI 🤦

Ideally they should not depend on attributes from within primer components, but it's common :(

Reply:
> Or are we worried this could also break tests in dotcom?

Only one way to find out 😓 Going to run the integration tests for this PR